plugins/linuxplugin/l3plugin/data_resync.go

```diff
@@ -64,36 +70,236 @@ func (plugin *LinuxRouteConfigurator) Resync(routes []*l3.LinuxStaticRoutes_Rout
+			return nil, fmt.Errorf("failed to parse destination IP address %s: %v", nbRoute.DstIpAddr, err)
+		}
+		linuxRts, err := netlink.RouteGet(dstNetIP.IP)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read linux route %s using address %s: %v",
+				nbRoute.Name, nbRoute.DstIpAddr, err)
+		}
+		if linuxRts != nil {
+			linuxRoutes = append(linuxRoutes, linuxRts...)
+		}
+	}
+	// Look for routes using interface
+	if nbRoute.Interface != "" {
+		link, err := netlink.LinkByName(nbRoute.Interface)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read interface %s: %v", nbRoute.Interface, err)
+		}
+		linuxRts, err := netlink.RouteList(link, netlink.FAMILY_ALL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read linux route %s using interface %s: %v",
+				nbRoute.Name, nbRoute.Interface, err)
+		}
+		if linuxRts != nil {
+			linuxRoutes = append(linuxRoutes, linuxRts...)
 		}
 	}
 
-	// Dump pre-existing not managed arp entries
-	err := plugin.LookupLinuxRoutes()
+	if len(linuxRoutes) == 0 {
+		plugin.Log.Debugf("Equivalent for route %s was not found", nbRoute.Name)
+	}
+
+	return linuxRoutes, nil
+}
+
+// Compare all route parameters and returns true if routes are equal, false otherwise
+func (plugin *LinuxRouteConfigurator) isRouteEqual(rtIdx int, nbRoute, linuxRt *l3.LinuxStaticRoutes_Route) bool {
+	// Common fields (interface, gateway)
+	if nbRoute.Interface != linuxRt.Interface {
+		plugin.Log.Debugf("Linux route %d: interface is different (NB: %s, Linux: %s)",
+			rtIdx, nbRoute.Interface, linuxRt.Interface)
+		return false
+	}
+	if nbRoute.GwAddr != linuxRt.GwAddr {
+		plugin.Log.Debugf("Linux route %d: gateway is different (NB: %s, Linux: %s)",
+			rtIdx, nbRoute.GwAddr, linuxRt.GwAddr)
+		return false
+	}
+	// Default route
+	if nbRoute.Default {
+		if !linuxRt.Default {
+			plugin.Log.Debugf("Linux route %d: NB route is default, but linux route is not", rtIdx)
+			return false
+		}
+		if nbRoute.Metric != linuxRt.Metric {
+			plugin.Log.Debugf("Linux route %d: metric is different (NB: %s, Linux: %s)",
+				rtIdx, nbRoute.Metric, linuxRt.Metric)
+			return false
+		}
+		return true
+	}
+	// Static route
+	_, nbIPNet, err := net.ParseCIDR(nbRoute.DstIpAddr)
 	if err != nil {
-		errs = append(errs, err)
+		plugin.Log.Error(err)
+		return false
+	}
+	if nbIPNet.IP.String() != linuxRt.DstIpAddr {
+		plugin.Log.Debugf("Linux route %d: destination address is different (NB: %s, Linux: %s)",
+			rtIdx, nbIPNet.IP.String(), linuxRt.DstIpAddr)
+		return false
+	}
+	if nbRoute.SrcIpAddr != linuxRt.SrcIpAddr {
+		plugin.Log.Debugf("Linux route %d: source address is different (NB: %s, Linux: %s)",
+			rtIdx, nbRoute.SrcIpAddr, linuxRt.SrcIpAddr)
+		return false
+	}
+	// If NB scope is nil, set scope type LINK (default value)
+	if nbRoute.Scope == nil {
+		nbRoute.Scope = &l3.LinuxStaticRoutes_Route_Scope{
+			Type: l3.LinuxStaticRoutes_Route_Scope_LINK,
+		}
+	}
+	if nbRoute.Scope != nil && linuxRt.Scope != nil {
```

Comment:
You could combine this with previous if using `} else if linuxRt.Scope != nil {`.